src/Math-Matrix/PMTensor.class.st

```diff
@@ -0,0 +1,152 @@
+Class {
+	#name : #PMTensor,
+	#superclass : #Object,
+	#instVars : [
+		'array',
+		'shape',
+		'first',
+		'strides'
+	],
+	#category : #'Math-Matrix'
+}
+
+{ #category : #'instance creation' }
+PMTensor class >> fromNestedArray: anArray [
+
+	^ self new array: anArray flattened withShape: (self shape:anArray)
+]
+
+{ #category : #'instance creation' }
+PMTensor class >> newWith: anInteger [
+
+	^ self new array: {anInteger} withShape: #(  )
+]
+
+{ #category : #accessing }
+PMTensor class >> shape: anArray [
+
+anArray isArray ifFalse:[^#()]
+ifTrue:[
+	^ {anArray size}, (self shape: anArray first)
+	]
+
+]
+
+{ #category : #comparing }
+PMTensor >> = aTensor [
+
+	^ array = aTensor array & (first = aTensor first)
+	  & (strides = aTensor strides) & (shape = aTensor shape)
+]
+
+{ #category : #accessing }
+PMTensor >> array [ 
```

Comment:
This method looks like an accessor, but I don't think it should be one. The use of an array to store the elements is an implementation detail. What users might want is a conversion to a (flat) array, which I'd then call `asArray`. I'd then rename the instance variable `array` to `elements` and have an accessor of that name for internal use (marked "private").